core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -208,14 +263,30 @@ public HttpResponse doCompleteInstall() throws IOException, ServletException {
         return HttpResponses.okJSON();
     }
     
-    static void completeSetup(Jenkins jenkins) {
+    static void completeSetup(Jenkins jenkins) throws IOException {
         jenkins.setInstallState(InstallState.INITIAL_SETUP_COMPLETED);
         InstallUtil.saveLastExecVersion();
+        UpgradeWizard.completeUpgrade(jenkins);
         // Also, clean up the setup wizard if it's completed
         jenkins.setSetupWizard(null);
     }
     
     /**
+     * Handle security configuration
+     */
+    public HttpResponse doSaveSecurityConfig(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, FormException {
+        jenkins.checkPermission(Jenkins.ADMINISTER); // this is redundant, maybe
+        GlobalSecurityConfiguration securityConfig = (GlobalSecurityConfiguration)jenkins.getDynamic("configureSecurity");
+        boolean configureSuccess = securityConfig.configure(req, req.getSubmittedForm());
+        if(configureSuccess) {
+            jenkins.save();
+            jenkins.setInstallState(InstallState.CONFIGURE_SECURITY.getNextState());
+            return HttpResponses.okJSON();
+        }
+        return HttpResponses.errorJSON("Unable to configure security");
```

Comment:
@kzantow Yeah, I'm not exactly sure where this was originally touched, I'm just seeing it for the first time now and it's making my Spidey Sense tingle.  Was this code path present before we added the SetupWizard somehow?